jira_bot.py

```diff
@@ -0,0 +1,160 @@
+""" jira_bot.py - Bot able to create jira-issues from pull-requests
+"""
+import argparse
+import os
+import sys
+import yaml
+
+from jira import JIRA
+
+JIRA_SERVER = os.getenv("JIRA_SERVER", "https://issues.redhat.com")
+DEFAULT_PROJECT_KEY = os.getenv("DEFAULT_PROJECT_KEY", "HMS")
+DEFAULT_ISSUE_TYPE = os.getenv("DEFAULT_ISSUE_TYPE", "Task")
+DEFAULT_COMPONENT = os.getenv("DEFAULT_COMPONENT", "Image Builder")
+
+
+def load_assignee_mapping(file_path):
+    """
+    Load GitHub-to-Jira username mappings from a YAML file.
+    """
+    try:
+        with open(file_path, 'r') as yaml_file:
+            data = yaml.safe_load(yaml_file)
+            return data.get('assignees', {})
+    except Exception as e:
+        print(f"Error loading YAML file '{file_path}': {e}", file=sys.stderr)
+        return {}
+
+
+def get_jira_username(github_nick, mapping):
+    """
+    Find the Jira username corresponding to a GitHub nickname.
+    """
+    return mapping.get(github_nick, None)
+
+
+def is_epic_issue(jira, issue_key):
+    """
+    Check if a Jira issue exists and is of the type 'Epic'.
+    """
+    try:
+        # Get the issue
+        print(f"Check if issue '{issue_key}' exists.", file=sys.stderr)
+        issue = jira.issue(issue_key)
+
+        # Check if the issue type is 'Epic'
+        print(f"Check if issue '{issue_key}' is an Epic.", file=sys.stderr)
+        if issue.fields.issuetype.name.lower() == 'epic':
+            return True
+        else:
+            return False
+    except Exception as e:
+        print(f"Error: {e}", file=sys.stderr)
+        return False
+
+
+# pylint: disable=too-many-arguments
+def create_jira_task(token, project_key, summary, description, issue_type, epic_link, component, assignee, story_points):
+    """
+    create_jira_task creates a jira issue with the given parameter
+    """
+    options = {
+        'server': JIRA_SERVER,
+        'headers': {
+            'Authorization': f'Bearer {token}'
+        }
+    }
+    try:
+        jira = JIRA(options=options)
+        print("Connected to Jira successfully using a personal access token.", file=sys.stderr)
+    # pylint: disable=broad-exception-caught
+    except Exception as e:
+        print(f"🔴 Failed to connect to Jira: {e}", file=sys.stderr)
+        return
+
+    # Check if Epic exists
+    if is_epic_issue(jira, epic_link) is False:
+        print(f"🔴 The Jira issue '{epic_link}' does not exist or is not of issuetype Epic.", file=sys.stderr)
+        sys.exit(1)
```

Comment:
`--epic-link` is currently optional, but if it's not given, the script will always `exit(1)` because of this conditional block. So either let's mark it as required (that's my suggestion), or alter the logic so not having an epic is fine.

Reply:
The thinking was that we'd always supply an Epic in the GitHub Action, but for local command line use it may not be required/desired.